corehq/apps/domain/views/settings.py

```diff
@@ -367,6 +367,7 @@ def post(self, request, *args, **kwargs):
         config, _ = CaseSearchConfig.objects.update_or_create(domain=self.domain, defaults={
             'enabled': request_json.get('enable'),
             'synchronous_web_apps': request_json.get('synchronous_web_apps'),
+            'sync_cases_on_form_entry': request_json.get('sync_cases_on_form_entry'),
```

Comment:
Similar to `synchronous_web_apps`/`case_search_synchronous_web_apps_for_domain`, I think the cache for `case_search_sync_cases_on_form_entry_enabled_for_domain` needs to be cleared below this.

Reply:
[5252a4c](https://github.com/dimagi/commcare-hq/pull/32554/commits/5252a4c174dfb88b4c729548b6b248a3ab81d350)